pkg/core/types.go

```diff
@@ -42,3 +42,17 @@ type Object interface {
 	runtime.Object
 	metav1.Object
 }
+
+// ConfigField is used to contain information about a specific field within
+// a ConfigMap/Secret to control how Wave uses that field.
+type ConfigField struct {
+	optional bool
+}
+
+// ConfigObject is used as a container of an "Object" along with metadata
+// that Wave uses to determine what to use from that Object.
+type ConfigObject struct {
```

Comment:
Same question, If this only has private fields, does it need to be public? Where is it being used outside of this package?

Reply:
Good point - will change.